pyccel/codegen/printing/fcode.py

```diff
@@ -3125,7 +3130,7 @@ def _print_PyccelPow(self, expr):
 
     def _print_PyccelAdd(self, expr):
         if isinstance(expr.dtype, StringType):
-            return '//'.join('trim('+self._print(a)+')' for a in expr.args)
+            return ' // '.join(self._print(a) for a in expr.args)
```

Comment:
Was `trim` necessary before this PR?

Reply:
No. Trim was used in the original implementation written by Ahmed and Said. This implementation declared strings as fixed length objects so trim was necessary to avoid an excessive number of spaces. However it had the downside of preventing us from doing:
```python
a = 'hello '
b = 'world'
c = a + b
```
with the old implementation with `trim`, `c`would have contained `'helloworld'` instead of `'hello world'`.

The declaration of strings was changed in #2063 ( 6a1c01b888 ) they are now allocatables so the length matches the Python length and `trim` serves no purpose